### PR TITLE
feat: Add Typescript types

### LINKED ISSUE #67: Add types
Hey! Could we add these types to the repository so TS users don't have to define them manually? 😄 

```ts
declare module '@cypress/webpack-preprocessor' {
  import * as webpack from 'webpack';

  type FilePreprocessor = (file: Cypress.FileObject) => string | Promise<string>;

  namespace _default {
    export type Options = { webpackOptions: webpack.Configuration };
  }

  const _default: (options: _default.Options) => FilePreprocessor;

  export = _default;
}
```

### PR DESCRIPTION
**Pull request by NickBolles**
Fixes #67

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,6 +62,7 @@
   "files": [
     "*.js"
   ],
+  "types": "index.d.ts",
```
Comment (OliverJAsh) [inline package.json:65]: `index.d.ts` is not included in `files`, so it's not part of the npm package:

https://unpkg.com/@cypress/webpack-preprocessor/

Comment (chrisbreiding) [reply package.json:65]: D'oh! I'll get that fixed and a new release out right away.

Comment (chrisbreiding) [reply package.json:65]: Fixed in [`5.1.1`](https://github.com/cypress-io/cypress-webpack-preprocessor/releases/tag/v5.1.1)

Comment (NickBolles) [reply package.json:65]: I could have sworn the types field was included. Main is, but types aren't I guess :(
https://docs.npmjs.com/files/package.json

**Pull request merged**